src/core/expr/fexpr_sumprod.cc

```diff
@@ -0,0 +1,131 @@
+//------------------------------------------------------------------------------
+// Copyright 2022 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#include "column/const.h"
+#include "column/sumprod.h"
+#include "documentation.h"
+#include "expr/fexpr_func.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "python/xargs.h"
+#include "stype.h"
+
+namespace dt {
+namespace expr {
+
+
+template <bool SUM>
+class FExpr_SumProd : public FExpr_Func {
+  private:
+    ptrExpr arg_;
+
+  public:
+    FExpr_SumProd(ptrExpr &&arg)
+        : arg_(std::move(arg)) {}
+
+    std::string repr() const override {
+      std::string out = SUM? "sum" : "prod";
+      out += '(';
+      out += arg_->repr();
+      out += ')';
+      return out;
+    }
+
+
+    Workframe evaluate_n(EvalContext &ctx) const override {
+      Workframe wf = arg_->evaluate_n(ctx);
+      Groupby gby = ctx.get_groupby();
+      if (!gby) {
+        gby = Groupby::single_group(wf.nrows());
+      }
+      
+      Workframe outputs(ctx);
+      for (size_t i = 0; i < wf.ncols(); ++i) {
+         Column coli = evaluate1(wf.retrieve_column(i), gby);
+         outputs.add_column(std::move(coli), wf.retrieve_name(i), Grouping::GtoONE);
+      }
```

Comment:
So we don't have any shortcuts for the grouped columns?

Reply:
is there a benefit to that? it will be reduced, so the same steps should be involved (`evaluate1`)

Reply:
Yes, please take a look at my comments below.